dandi/download.py

```diff
@@ -1238,49 +1239,65 @@ def get_status(self, report_done: bool = True) -> dict:
 
     def feed(self, path: str, status: dict) -> Iterator[dict]:
         keys = list(status.keys())
-        self.files.setdefault(path, DownloadProgress())
         size = status.get("size")
         if size is not None:
             if not self.yielded_size:
                 # this thread will yield
                 self.yielded_size = True
                 yield {"size": self.zarr_size}
         if status.get("status") == "skipped":
-            self.files[path].state = DLState.SKIPPED
+            self.files_fed += 1
+            self.file_states[DLState.SKIPPED] += 1
+            try:
+                self.total_downloaded -= self.downloading.pop(path).downloaded
+            except KeyError:
+                pass
             # Treat skipped as "downloaded" for the matter of accounting
             if size is not None:
-                self.files[path].downloaded = size
+                self.total_downloaded += size
                 self.maxsize += size
             yield self.get_status()
         elif keys == ["size"]:
-            self.files[path].size = size
-            self.maxsize += status["size"]
-            if any(s.state is DLState.DOWNLOADING for s in self.files.values()):
+            self.files_fed += 1
+            self.file_states[DLState.STARTING] += 1
+            assert size is not None
+            self.downloading[path] = DownloadProgress(size=size)
```

Comment:
```suggestion
            assert path not in self.downloading
            self.downloading[path] = DownloadProgress(size=size)
```

correct? 
since otherwise we would be double counting for `self.files_fed` with below.

Reply:
Done in 6277f78f.